### PR TITLE
Created RobotSaveState & GlobalStateTest

### PR DESCRIPTION
**Pull request by Thorny128**

### CODE REVIEW

File: core/src/main/java/team/techtigers/core/RobotSaveState.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * @param state The new value to be saved
+     */
+    public void setState(HashMap<String, Object> state) {
+        this.state = state;
+    }
+
+    /**
+     * Resets the state to be null
+     */
+    public void reset() {
```
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java]: Make this static

File: core/src/main/java/team/techtigers/core/RobotSaveState.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        return instance;
+    }
+
+    /**
+     * Gets the current robot state
+     * @return The current robot state
+     */
+    public HashMap<String, Object> getState() {
```
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java:33]: So this should take an input of a String, and the return should be something like state.get(key). We aren't returning the whole state, just a specific value requested

File: core/src/main/java/team/techtigers/core/RobotSaveState.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            return state;
+        } catch (NullPointerException e){
+            System.out.println("Error: No state has been set");
+            System.out.println("Returning null instead");
+            return null;
+        }
+    }
+
+    /**
+     * Sets the current robot state, requires a string for identification
```
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java:44]: This should take an input of. a String and an Object, and the method should do state.set(String, Object). There shouldn't be a whole new map being set

File: core/src/main/java/team/techtigers/core/RobotSaveState.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * Gets the current robot state
+     * @return The current robot state
+     */
+    public HashMap<String, Object> getState() {
+        try {
+            return state;
+        } catch (NullPointerException e){
+            System.out.println("Error: No state has been set");
```
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java:37]: Don't do this. Throw an IllegalStateException and put the message inside the exception

ex. 
``` throw new IllegalStateException("No state has been set") ```

**Review by greenbear26 — CHANGES_REQUESTED** [review]

File: core/src/main/java/team/techtigers/core/RobotSaveState.java (hunk 2 above)
Comment (Thorny128) [reply core/src/main/java/team/techtigers/core/RobotSaveState.java:33]: Should have thought of that, thanks for clarifying

File: core/src/main/java/team/techtigers/core/RobotSaveState.java
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java:1]: Please write unit tests for this file

**Review by greenbear26 — CHANGES_REQUESTED** [review]

File: core/src/main/java/team/techtigers/core/RobotSaveState.java (hunk 1 above)
Comment (Thorny128) [reply core/src/main/java/team/techtigers/core/RobotSaveState.java]: Oops

File: core/src/main/java/team/techtigers/core/RobotSaveState.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Gets the current robot state
+     * @return The current robot state
+     */
+    public Object getState(String key) {
+        if (!state.containsKey(key)) {
+            throw new IllegalStateException("No state has been set for key: " + key);
+        }
+            return state.get(key);
```
Comment (greenbear26) [inline core/src/main/java/team/techtigers/core/RobotSaveState.java:39]: Fix indentation here

Comment (greenbear26) [reply core/src/main/java/team/techtigers/core/RobotSaveState.java:39]: Actually, you can put this as an else condition like in the method below

**Review by greenbear26 — CHANGES_REQUESTED** [review]
Small things

**Review by windigo216 — APPROVED** [review]

**Pull request merged**